extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/NoArgsConstructorProcessor.java

```diff
@@ -127,6 +112,27 @@ public boolean skipValidation(InjectionTargetInfo target, ValidationRule rule) {
         }
     }
 
+    private Collection<ClassInfo> collectTargetClass() {
```

Comment:
Modified because to be able to create a default `noArgConstructor` for `AppScope` and `RequestScope` once. Otherwise, TCK tests in `HasRequestScopeTest` and `HasAppScopeTest` were failing with a `<cinit>` error.

Reply:
We would need validation from @mkouba on this one

Reply:
It wouldn't hurt to describe what's actually changed and why? ;-)

Reply:
**Why**
While working on this PR, I noticed the following error:
```
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl1.<clinit>(ApplicationImpl1.zig:205)
	... 112 more
Caused by: java.lang.ClassFormatError: Duplicate method name "<init>" with signature "()V" in class file org/eclipse/microprofile/rest/client/tck/cditests/HasAppScopeTest$MyAppScopedApi
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:642)
	at io.quarkus.runner.RuntimeClassLoader.findClass(RuntimeClassLoader.java:217)
	at io.quarkus.runner.RuntimeClassLoader.loadClass(RuntimeClassLoader.java:163)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.eclipse.microprofile.rest.client.tck.cditests.HasAppScopeTest_MyAppScopedApi_4a90cb1eedbe1d7e3f457833a0ee883318179f2e_Synthetic_Bean.<init>(HasAppScopeTest_MyAppScopedApi_4a90cb1eedbe1d7e3f457833a0ee883318179f2e_Synthetic_Bean.zig:120)
	at io.quarkus.arc.setup.Default_ComponentsProvider.getComponents(Default_ComponentsProvider.zig:222)
	at io.quarkus.arc.ArcContainerImpl.<init>(ArcContainerImpl.java:84)
	at io.quarkus.arc.Arc.initialize(Arc.java:18)
	at io.quarkus.arc.runtime.ArcDeploymentTemplate.getContainer(ArcDeploymentTemplate.java:33)
	at io.quarkus.deployment.steps.ArcAnnotationProcessor$build7.deploy_0(ArcAnnotationProcessor$build7.zig:53)
	at io.quarkus.deployment.steps.ArcAnnotationProcessor$build7.deploy(ArcAnnotationProcessor$build7.zig:141)
	at io.quarkus.runner.ApplicationImpl1.<clinit>(ApplicationImpl1.zig:191)
	... 112 more
``` 
When running `rest-client` tck tests notably when running and `HasAppScopeTest` or `HasRequestScopeTest`.  My understanding is, we tried to generate a no arg constructor twice and looking at the modified code, an `HashSet` is used while `ClassInfo` does not declare an `hashCode` method leading to `MyAppScopedApi.class` being recorded twice. 
  
**What changed**
Record a normal scoped bean class once based on its  `DotName` which has a `hashCode`. The  `DotName` is provided by `ClassInfo.name()`. 
 
Is that the way to go about it? Open to changing this to something different. @mkouba @geoand

Reply:
Seems reasonable to me, @mkouba WDYT?

Reply:
@machi1990 thanks for explanation. When looking at the test I'm a bit confused. So the spec states _"Interfaces are assumed to have a scope of `@Dependent` unless there is another scope defined on the interface. Implementations are expected to support all of the built in scopes for a bean."_. But from CDI POV interfaces are NOT beans and scope annotations are ignored there. I need to think about this a bit more...

Reply:
Anyway, to fix the problem with duplicit no-args constructor - we should just ignore interfaces here: https://github.com/quarkusio/quarkus/blob/master/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/NoArgsConstructorProcessor.java#L144 using `Modifier.isInterface(annotationInstance.target().asClass().flags())`.

Reply:
@machi1990 otherwise the PR looks good! So pls revert the changes in `NoArgsConstructorProcessor` and just ignore the interfaces there.

Reply:
@mkouba @geoand thanks for your analysis.

> Modifier.isInterface(annotationInstance.target().asClass().flags())

I'll revert this part and use this instead.

Reply:
@mkouba did all the meaningful analysis 😆

Reply:
> @mkouba @geoand thanks for your analysis.
> 
> > Modifier.isInterface(annotationInstance.target().asClass().flags())
> 
> I'll revert this part and use this instead.

This had been addressed. @mkouba @geoand

Reply:
I'll let @mkouba review since he was the one that found the problem